lib/auth/auth_with_roles.go

```diff
@@ -103,9 +103,8 @@ func (a *ServerWithRoles) actionWithContext(ctx *services.Context, namespace, re
 	for _, verb := range verbs {
 		errs = append(errs, a.context.Checker.CheckAccessToRule(ctx, namespace, resource, verb, false))
 	}
-	// Convert generic aggregate error to AccessDenied.
 	if err := trace.NewAggregate(errs...); err != nil {
-		return trace.AccessDenied(err.Error())
+		return err
```

Comment:
This was here to make `trace.IsAccessDenied` work, but is no longer necessary now that aggregate errors support Is/As since https://github.com/gravitational/trace/pull/77